### ISSUE TITLE
CodeQL Warning in LAMMPS

### ISSUE THREAD

**Issue by akohlmey** [Colvars/colvars#422]
Static code analysis of a full LAMMPS compile (with COLVARS included) resulted in this report.

https://github.com/lammps/lammps/security/code-scanning/448

Please let me know if you consider this worth addressing or whether I should just dismiss it.

**Comment by giacomofiorin** [comment]
I don't believe this would ever be an actual security issue given the uses of that function. 

I'd ignore the warning but I've also been thinking about getting rid of that function (one fewer virtual function to worry about), so I can address it in a small PR to LAMMPS (there is also #419 since my last PR to LAMMPS that may be worth including).

https://github.com/Colvars/colvars/blob/83dc68fc0bc03c65ddcfbb70086b3cc24e502ee8/src/colvarproxy.cpp#L642-L646
https://github.com/Colvars/colvars/blob/83dc68fc0bc03c65ddcfbb70086b3cc24e502ee8/src/colvarproxy.cpp#L682-L700
